lisp/core/analyze.el

```diff
@@ -100,19 +105,21 @@ Argument LEVEL and MSG are data from the debug log signal."
     (dolist (file files)
       (eask--silent-error
         (eask--save-load-eask-file file
+            (push file checked-files)
+            ;; also count files with errors in the total count
             (push file checked-files))))
 
     ;; Print result
     (eask-msg "")
-    (cond ((and (eask-json-p)  ; JSON format
-                (or eask-analyze--warnings eask-analyze--errors))
-           (setq content
-                 (eask-analyze--pretty-json (json-encode
-                                             `((warnings . ,eask-analyze--warnings)
-                                               (errors   . ,eask-analyze--errors)))))
+    (cond ((eask-json-p)  ; JSON format
+           ;; Fill content with result.
+           (when (or eask-analyze--warnings eask-analyze--errors)
+             (setq content (json-encode
```

Comment:
The pretty print `eask-analyze--pretty-json ` is probably still better?

Reply:
There needs to be a way to separate JSON from plain text in tests. See the function `tryJSON` in `analyze.test.js`. It wont work if the JSON is pretty printed.

I'm assuming that the output from `analyze --json` is just for flycheck/flymake and so it doesn't need to be pretty printed.

The other alternative would be to write something like this https://stackoverflow.com/a/54608716, which is kind of messy and likely fragile.

Reply:
What if we convert the pretty print result to a single line result in the `tryJSON` function? 🤔 

A printed result is a better UX, since I seldom read the JSON output directly from the terminal. Additionally, it's easier to debug and add new content to the output. :)

Reply:
I mean, in general, you can change the code in `analyze.el` so that `analyze --json` prints output that causes the test to fail while `flycheck-eask` is still working.